File: 000401-000600/000412.FizzBuzz/412.go
package main

import (
	"fmt"
	"strconv"
)

// 方法一：用时 8ms
// 执行用时超过了 9%
// 消耗内存超过了 30%
func fizzBuzz(n int) []string {
	ret := []string{}
	for i := 1; i <= n; i++ {
		if i % 3 != 0 && i % 5 != 0 {
			ret = append(ret, strconv.Itoa(i))
		} else if i % 3 == 0 && i % 5 != 0 {
			ret = append(ret, "Fizz")
		} else if i % 3 != 0 && i % 5 == 0 {
			ret = append(ret, "Buzz")
		} else {
			ret = append(ret, "FizzBuzz")
		}
	}
	return ret
}

func main ()  {
	fmt.Println(fizzBuzz(15))
}

File: 000401-000600/000412.FizzBuzz/412.js
/**
 * @param {number} n
 * @return {string[]}
 */

 // 方法一：用时 68ms
 // 执行用时超过了 95%
 // 消耗内存超过了 13%
var fizzBuzz = function(n) {
  let ret = []
  for (var i = 1; i <= n; i++) {
    if (i % 3 && i % 5) {
      ret.push(i.toString())
    } else if (!(i % 3) && i % 5) {
      ret.push('Fizz')
    } else if (i % 3 && !(i % 5)) {
      ret.push('Buzz')
    } else {
      ret.push('FizzBuzz')
    }
  }
  return ret
};

console.log(fizzBuzz(15))
